src/main/java/appeng/core/sync/packets/PacketCraftingTreeData.java

```diff
@@ -51,24 +64,68 @@ public PacketCraftingTreeData(final CraftingJobV2 job) {
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
+        // Split into chunks
+        final int chunkCount = (int) Platform.ceilDiv(output.readableBytes(), CHUNK_SIZE);
+        final ArrayList<PacketCraftingTreeData> chunks = new ArrayList<>(chunkCount);
+        for (int chunk = 0; chunk < chunkCount; chunk++) {
+            final int start = CHUNK_SIZE * chunk;
+            final int end = Math.min(start + CHUNK_SIZE, output.readableBytes());
+            final int len = end - start;
+            chunks.add(new PacketCraftingTreeData(output.slice(start, len), chunk, chunkCount));
+        }
         if (AEConfig.instance.isFeatureEnabled(AEFeature.DebugLogging)) {
             AELog.info(
-                    "Crafting tree packet raw size %d, compressed %d",
+                    "Crafting tree packet raw size %d, compressed %d, chunk count %d",
                     jobData.writerIndex(),
-                    output.readableBytes());
+                    output.readableBytes(),
+                    chunks.size());
         }
-        this.configureWrite(output);
+        return chunks;
     }
 
+    // Store partially received packets client-side
+    private static final WeakHashMap<EntityPlayer, ByteBuf[]> chunkStorage = new WeakHashMap<>();
+
     @Override
     @SideOnly(Side.CLIENT)
     public void clientPacketData(final INetworkInfo network, final AppEngPacket packet, final EntityPlayer player) {
         if (receivedData == null) {
             return;
         }
+        final int chunkId = receivedData.readInt();
+        final int totalChunks = receivedData.readInt();
+        if (totalChunks <= 0 || chunkId < 0 || chunkId >= totalChunks) {
+            AELog.warn("Invalid chunked crafting tree packet received from server: Chunk %d/%d", chunkId, totalChunks);
+            return;
+        }
+        if (totalChunks == -1) {
```

Comment:
Is this branch useless? returns above if totalChunks <= 0

Reply:
Oops, looks like I left my debug edit in for testing the chunking code, this is meant to be ==1 not -1